### PR TITLE
snap: qemu: trim the list of QEMU devices 

### LINKED ISSUE #682: snap: trim the list of QEMU devices
Use kconfig to trim the list of QEMU devices

### PR DESCRIPTION
**Pull request by devimc**
Trim the list of devices to reduce the attack surface and binary size

fixes #682

### CODE REVIEW

**Comment by devimc** [comment]
/test

CI is happy, let's try again

**Comment by jodh-intel** [comment]
Hi @devimc - can you explain how `config-devices.mak` interacts with https://github.com/kata-containers/packaging/blob/master/scripts/configure-hypervisor.sh? aiui, running QEMU's `configure` script sets certain `CONFIG_*` options, so how do they get combined with `config-devices.mak`?

**Comment by devimc** [comment]
@jodh-intel `configure-hypervisor.sh` + `./configure` disable/enable host's features (i.e kvm, xen, etc) and devices (i.e ac97, e1000, usb audio, vga, etc). Unfortunately there are no configure options to disable specific devices, for example there is **no** an option to disable e1000 (--disable-e1000), the only way to disable specific devices is editing `config-devices.mak` (CONFIG_E1000_PCI=n). Having said that, `configure-hypervisor.sh` + `./configure` will be used to disable/enable host's features  and `config-devices.mak` to disable/enable devices.

**Comment by devimc** [comment]
/test

testing QEMU 4.1.0-rc2 + new cpu topology + `qemu-mini` (kconfig)

**Comment by devimc** [comment]
CI is happy, but I'd like to see the metrics-ci failing :smile:  https://github.com/kata-containers/tests/pull/1861

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -296,7 +297,7 @@ parts:
         *)
           branch="$(curl -sSL ${versions_url} | ${yq} r - assets.hypervisor.qemu.tag)"
           url="$(curl -sSL ${versions_url} | ${yq} r - assets.hypervisor.qemu.url)"
-          patch_dir="${SNAPCRAFT_STAGE}/qemu/patches/4.0.x"
+          patch_dir="${SNAPCRAFT_STAGE}/qemu/patches/$(echo ${branch} | cut -d. -f1-2 | tr -d v).x"
```
Comment (jodh-intel) [inline snap/snapcraft.yaml]: Nice.

File: qemu/default-configs/i386-softmmu.mak [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# Boards
+CONFIG_ISAPC=n
+CONFIG_I440FX=y
+CONFIG_Q35=y
+CONFIG_ACPI_PCI=y
```
Comment (jodh-intel) [inline qemu/default-configs/i386-softmmu.mak]: NIt: Would be good if the config options in these blocks were sorted ;)

**Comment by devimc** [comment]
/test 
@jodh-intel change applied, thanks

**Comment by devimc** [comment]
moving changes to https://github.com/kata-containers/packaging/pull/688

**Pull request closed without merging**